server/src/main/java/io/crate/types/NumericType.java

```diff
@@ -123,21 +124,25 @@ public BigDecimal implicitCast(Object value) throws IllegalArgumentException, Cl
             return null;
         }
 
-        var mathContext = mathContext();
         BigDecimal bd;
         if (value instanceof BigDecimal bigDecimal) {
-            bd = bigDecimal.round(mathContext);
+            bd = bigDecimal;
         } else if (value instanceof String || value instanceof Float || value instanceof Double) {
-            bd = new BigDecimal(value.toString(), mathContext);
+            bd = new BigDecimal(value.toString());
         } else if (value instanceof Number number) {
-            bd = new BigDecimal(BigInteger.valueOf(number.longValue()), mathContext);
+            bd = new BigDecimal(BigInteger.valueOf(number.longValue()));
```

Comment:
These changes have no effect on test succession, can you please explain why they are needed? (or add tests verifying the change)

Reply:
Since we do rounding, just before returning the `bd` if `scale != null`, those are just extra steps of rounding that is done anyway afterwards. Nothing breaks, but since I cannot think of an counter example, (that might exist), which breaks with my change, maybe it's better to leave it as it was?

Reply:
> Since we do rounding, just before returning the bd if scale != null, those are just extra steps of rounding that is done anyway afterwards. 

Ah makes sense, not sure, fine for me to remove.

Reply:
on second thought, and checking again the `round` and the `new BigDecimal()`, I'd rather keep it as is to avoid breaking something. I will revisit and if 100% sure those additional operations are covered by just `setScale()` I'll remove in a separate PR.